### ISSUE TITLE
Double quote column names in postgres when using "include_missing_columns": true

### ISSUE THREAD

**Issue by Gitznik** [EqualExperts/dbt-unit-testing#170]
When using `"include_missing_columns": true` with my postgres data and one of the column names of the table to mock includes a reserved keyword (in my case `reference`), the test fails to compile. For context - the mocked source table contains the `dbt_utils.star` macro, in case that's relevant.

```
12:26:22  Database Error in test test_manual_overrides (tests/unit/test_manual_overrides.sql)
12:26:22    syntax error at or near "references"
12:26:22    LINE 14: ...col1,col2,col3,references...
```

Looking at the query that is written based on the test we can see, that the column names that are selected are not quoted, leading to the above error.

This is the specific part of the query causing problems:

```sql
select
	col1,
	col2,
	col3,
	references
from
(
	select
		*
	from
		"db"."raw"."table_name"
	where
		false) as m2
```

Any way to change this to always quote the column names?

This is how dbt_utils does it, see [here](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/star.sql#L31C1-L35C28):
```jinja
                {%- if quote_identifiers -%}
                    {{ adapter.quote(col)|trim }} {%- if prefix!='' or suffix!='' %} as {{ adapter.quote(prefix ~ col ~ suffix)|trim }} {%- endif -%}
                {%- else -%}
                    {{ col|trim }} {%- if prefix!='' or suffix!='' %} as {{ (prefix ~ col ~ suffix)|trim }} {%- endif -%}
                {% endif %}
```

**Comment by rensoostenbach** [comment]
I would also like to have this implemented for a different use case. My source table columns contain spaces in them which is currently not handled well and would be fixed by quoting the column name.

Ofcourse another solution would be to get rid of spaces in the column names, but that is not always in our control.

**Comment by Gitznik** [comment]
@psousa50 any chance we can get https://github.com/EqualExperts/dbt-unit-testing/pull/153 merged to resolve this? Sorry to mention you directly but I'd love to see this resolved. Thanks 🙏

**Comment by psousa50** [comment]
I think this was fixed in v0.3.4, could you please check?

Thank you!

**Comment by psousa50** [comment]
That's great @rensoostenbach , thank you for your feedback 👍
